apps/src/lib/levelbuilder/code-docs-editor/AllCodeDocs.jsx

```diff
@@ -0,0 +1,104 @@
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import ProgrammingEnvironmentsTable from './ProgrammingEnvironmentsTable';
+import ProgrammingExpressionsTable from './ProgrammingExpressionsTable';
+import {TextLink} from '@dsco_/link';
+
+export default function AllCodeDocs({programmingEnvironments, allCategories}) {
+  const [ideSelected, setIdeSelected] = useState(true);
+  const ideSelectButtonStyle = {
+    ...styles.button,
+    ...styles.leftButton,
+    ...(ideSelected && styles.selectedButton)
+  };
+  const docsSelectButtonStyle = {
+    ...styles.button,
+    ...styles.rightButton,
+    ...(!ideSelected && styles.selectedButton)
+  };
+
+  return (
+    <div>
+      <h1>All Code Documentation</h1>
+      <div style={styles.buttonRow}>
+        <div>
+          <button
+            type="button"
+            style={ideSelectButtonStyle}
+            onClick={() => setIdeSelected(true)}
+          >
+            IDEs
+          </button>
+          <button
+            type="button"
+            style={docsSelectButtonStyle}
+            onClick={() => setIdeSelected(false)}
+          >
+            Code Documentation
+          </button>
+        </div>
+        <div style={styles.newLinks}>
+          <TextLink
+            href="/programming_environments/new"
+            icon={<i className="fa fa-plus-circle" aria-hidden="true" />}
+            text="New IDE"
+            iconBefore
+            openInNewTab
+          />
+          <TextLink
+            href="/programming_expressions/new"
+            icon={<i className="fa fa-plus-circle" aria-hidden="true" />}
+            text="New Code Doc"
+            iconBefore
+            openInNewTab
+          />
+        </div>
+      </div>
+      <div>
+        {' '}
```

Comment:
is this extra space important, or did the autoformat add it?

Reply:
I didn't add it so it must've been autoformat. But I've seen this happen when I've got a bug then I forget to remove it, so I'll try to remove it.